### PR TITLE
DOCS-119 Add jet migration guide

### LINKED ISSUE #162: Docs: Feedback for Upgrading from IMDG and Jet Version 4.x
Hi, I have some feedback about [this page](https://github.com/hazelcast/hz-docs/edit/main/docs/modules/migrate/pages/upgrade.adoc)

```
The 3.12.x versions of Hazelcast Java client cannot work with Hazelcast 5.0; see the Client Migration section.
```

It is not just the 3.12.x version of the Java client that cannot work with Hazelcast 5.0. Other non-Java clients listed above this sentence are also not able to work with 5.0.

### PR DESCRIPTION
**Pull request by JakeSCahill**
Fixes #162 and #163

### CODE REVIEW

**Review by vladoschreiner — CHANGES_REQUESTED** [review]
I can't locate the entry point to the Jet migration guide in the Netlify preview.

File: docs/modules/migrate/pages/upgrading-from-jet.adoc [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+```java
+HazelcastInstance hz = Hazelcast.newHazelcastInstance();
+JetService jet = hz.getJet();
+```
+--
+====
+
+== Enabling the Jet Engine
+
+In embedded mode (Hazelcast as a classic Java process), Jet is disabled by default. As a result, you cannot use Jet APIs or SQL unless you enable the Jet engine.
```
Comment (vladoschreiner) [inline docs/modules/migrate/pages/upgrading-from-jet.adoc:92]: Hazelcast always runs as a Java process. The embedded mode refers to "Hazelcast running in the JVMs of your application" as opposed to "Hazelcast running in its own standalone JVMs.

File: docs/modules/migrate/pages/upgrading-from-jet.adoc [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+Hazelcast {page-component-version} uses https://semver.org/[semantic versioning]:
+
+* MAJOR version when you make incompatible API changes
+* MINOR version when you add functionality in a backwards-compatible manner
+* PATCH version when you make backwards-compatible issue fixes.
+
+Hazelcast {page-component-version} is compatible only with the Job and Pipeline APIs of Jet 4.x. As a result, jobs that were running on Jet 4.x clusters can still be executed on Hazelcast {page-component-version} clusters.
+
+Hazelcast {page-component-version} *is not* compatible with any other aspects of Jet 4.x. You cannot upgrade using rolling upgrades and you cannot connect to the cluster using version 4.x of clients).
```
Comment (vladoschreiner) [inline docs/modules/migrate/pages/upgrading-from-jet.adoc:110]: Please be more specific:
- I can't do rolling upgrade, that is clear. I must do a down-time upgrade.
- Can I move the Job to from Jet 4 to Hz 5 by copying the job snapshot? Are jobs snapshot-compatible?
- Can I connect to Hz 5 cluster using Jet 4 client? Or do I have to upgrade the version of my client libs?

File: docs/modules/migrate/pages/upgrading-from-jet.adoc [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+** Azure discovery plugin
+** Hazelcast Hibernate
+** Hazelcast Spring
+** Default configuration files, including examples
+** Scripts
+** SQL
+** Jet extensions
+** Management center
+
+For a complete list of contents in each distribution, see xref:deploy:installing-upgrading.adoc#full-and-slim-packages[Full and Slim Packages].
```
Comment (vladoschreiner) [inline docs/modules/migrate/pages/upgrading-from-jet.adoc:138]: Please provide an action point. Does the distribution change require my attention?

File: docs/modules/migrate/pages/upgrading-from-jet.adoc [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+config.setClusterName("jet-dev");
+
+HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
+```
+--
+====
+
+=== InstanceConfig
+
+The link:https://jet-start.sh/javadoc/4.5/com/hazelcast/jet/config/InstanceConfig.html[`InstanceConfig` object] has been removed and all its methods have been added to the `JetConfig` object.
```
Comment (ufukyilmaz) [inline docs/modules/migrate/pages/upgrading-from-jet.adoc:74]: it's been deprecated and will be removed in a future release (it's not removed yet).

**Review by vladoschreiner — APPROVED** [review]

**Pull request merged**